ecl/eclcc/WORKUNITS.rst

```diff
@@ -0,0 +1,862 @@
+  </Result>
+  <Result name="Result 2" sequence="1">
+   <SchemaRaw xsi:type="SOAP-ENC:base64">
+    Result_2&#xe000;&#xe004;&#241;&#255;&#255;&#255;&#xe001;ascii&#xe000;&#xe001;ascii&#xe000;&#xe000;&#xe018;&#xe000;&#xe000;&#xe000;&#xe000;   </SchemaRaw>
+  </Result>
+ </Results>
+ 
+in our example there are two - the dataset of results and the text string "Done!".  The results of a query are
+associated with the workunit.  (They are currently saved in dali, but this may change in version 6.0.)
+
+Timings and statistics
+----------------------
+Any timings generated when compiling the query are included in the workunit dll::
+
+     <Statistics>
+      <Statistic c="eclcc"
+                 count="1"
+                 creator="eclcc"
+                 kind="SizePeakMemory"
+                 s="compile"
+                 scope="compile"
+                 ts="1428933081084000"
+                 unit="sz"
+                 value="27885568"/>
+     </Statistics>
+
+Other statistics and timings created when running the query are stored in the runtime copy of the workunit.
+(Statistics for graph elements are stored in a different format from global statistics, but the IWorkUnit interface
+ensures the implementation details are hidden.)
+
+Manifests
+---------
+
+It is possible to include other user-defined resources in the workunit dll - e.g., web pages, or dashboard layouts.
+I have to confess I don't understand them... ??Tony please provide some more information....!
+
+
+Stages of execution
+*******************
+Once a workunit has been compiled to a dll it is ready to be executed.  Execution can be triggered in different ways:
+
+* A query is submitted to a queue.  The query is compiled and eclccserver adds a workunit to a queue ready for execution.
+* A query is submitted and published.  Another request is then submitted to execute this previously compiled query.
+* A query is compiled as a stand alone executable.  The executable is then run.
+
+Most queries create persistent workunits in dali and then update those workunits with results as they are calculated, but
+for some roxie queries (e.g., in a production system) the execution workunits are transient.
+Whichever path is followed, the following walk-through details the main stages in executing a query, and the effect each
+of the query elements has.
+
+Workflow
+========
+When a workunit is ready to be run, the workflow controls the flow of execution.
+The workflow engine (common/workunit/workflow.cpp) is responsible for determining which workflow item should be
+executed next.  (The workflow for Thor and hThor jobs is coordinated by eclagent, while roxie includes the workflow
+engine in its process.  The eclscheduler also uses the workflow engine to process events and mark workflow items ready
+to be executed.)
```

Comment:
to be executed -> for execution
'.' should be outside the parens, the use of which at all should be rethought.  It is unusual for such to span more than a single sentence and long ones.